### ISSUE TITLE
Restaurant Revenue Prediction using Machine Learning

### ISSUE THREAD

**Issue by keshripritesh** [PRIYANSHU2026/DataSentience-AIML#768]
# 🍽️ Restaurant Revenue Prediction using Machine Learning

This project focuses on predicting restaurant revenue using a machine learning model, specifically a pre-trained **Random Forest** model. Users provide input features including location, cuisine type, rating, seating capacity, social media followers, chef experience in years, number of reviews, average review length, ambience score, service quality score, parking availability, weekend and weekday reservations, marketing budget in rupees, and average meal price in rupees.

The app preprocesses and transforms these inputs, encodes categorical variables, and applies log transformations where necessary. Built with **Streamlit**, the user-friendly interface allows users to easily input data and receive a revenue prediction. This helps restaurant owners make informed business decisions.

The project also demonstrates effective **data preprocessing, feature engineering, model training, and deployment** in a visually appealing web application.

---

## 📊 Dataset

The dataset used contains **~8000+ rows** and is sourced from Kaggle:

[[Restaurant Revenue Prediction Dataset](https://www.kaggle.com/datasets/anthonytherrien/restaurant-revenue-prediction-dataset)](https://www.kaggle.com/datasets/anthonytherrien/restaurant-revenue-prediction-dataset)

> The complete model training is in `restaurant_revenue_prediction.ipynb`. Please download the dataset and refer to the notebook after loading it.

The project also includes **stock price prediction** for **Indian Hotels Company (INDHOTEL.NS)** in `indian_hotel_stock_analysis.ipynb`.
`INDHOTEL.NS.csv` contains stock data from **Jan 1, 2012, to present**, collected from Yahoo Finance.

---

## 🧩 Methodology

The project follows a structured methodology:

1. **Data Collection**
2. **Data Preprocessing**
3. **Exploratory Data Analysis (EDA)**
   Using `matplotlib`, `pandas`, `seaborn`, and `plotly`.
   Plots include:

   * Pie charts
   * Histograms
   * Hexbin plots
   * Count plots
   * Correlation matrix
   * Regression plots
   * Scatter plots
   * Cross-validation plots
   * Bar plots

   > Refer to the `output/` folder for images and observations.
4. **Model Training and Evaluation**
   Models trained:

   * Random Forest ✅ (best accuracy)
   * Linear Regression
   * XGBoost
   * Decision Trees
5. **Inference / Deployment**
   Deployed using **Streamlit** to predict monthly restaurant revenue.

---

## 🛠️ Libraries Used

* **joblib**: Load and save models
* **plotly**: Interactive 2D & 3D visualizations
* **Matplotlib**: Plotting and visualizations
* **Pandas**: Data manipulation
* **NumPy**: Efficient numerical operations
* **Streamlit**: Web application GUI

---

## 🚀 How to Use

1. **Clone the repository**

```bash
git clone <repository_url>
```

2. **Install dependencies**

```bash
pip install -r requirements.txt
```

3. **Download and add pre-trained model**

* Link: [[rf_model.pkl](https://drive.google.com/file/d/1n0KeiJyfaoawFhewD3ze7rL9s3EgoxYk/view?usp=sharing)](https://drive.google.com/file/d/1n0KeiJyfaoawFhewD3ze7rL9s3EgoxYk/view?usp=sharing)
* Place `rf_model.pkl` in the same directory as `main.py`.

4. **Run the Streamlit app / script**

```bash
python main.py
```

5. **Input restaurant details** in the app and get the **predicted monthly revenue**.

---

## 🧑‍💻 Contributor

**Pritesh Ram Keshri**

---
